clang/test/CodeGenHLSL/builtins/dot-builtin.hlsl

```diff
@@ -6,7 +6,7 @@
 // CHECK: %conv2 = fptrunc double %hlsl.dot to float
 // CHECK: ret float %conv2
 float builtin_bool_to_float_type_promotion ( float p0, bool p1 ) {
-  return __builtin_hlsl_dot ( p0, p1 );
+  return __builtin_hlsl_dot ( (double)p0, (double)p1 );
```

Comment:
@farzonl I've quickly hacked this in place to satisfy the tests. The HLSL builtins use the same Sema APIs as the ones I'm interested in changing. Is the old behaviour something that needs preserved?

Reply:
From the checks it looks like you have just made the double more explicit. Codegen doesn't look different. So I'm fine with this change.

Also most of the type promotion rules for HLSL are enforced on the HLSL header. We are just using the builins as aliases for the cases we need to invoke an intrinsic.

Note to myself though these tests don't look correct and should probably be removed. DXC would have promoted bool to a float. Short of setting the builtin as `CustomTypeChecking` I don't think that can be achieved here. But again it does not matter because the promotion rules are handled on the hlsl `dot` api which only aliases to the builtin.

Reply:
Thanks for the reply. I think that make sense, thank you.

Note though that I had to make a change to what I think is a user-facing `pow` builtin [here](https://github.com/llvm/llvm-project/pull/119423/files#diff-fdbd201243000d44dd63e6af5706293c968588d762bed9e050d27e0abd6e80acR40), so this PR might still impact HLSL users directly.